### PR TITLE
Plate 4 Analysis

### PR DESCRIPTION
**Pull request by jenna-tomkinson**
# Plate 4 Analysis

In this PR, I perform the following analyses:

1. Generate UMAPs with the morphology space comparing between the hearts, treatments, and cell count per well.
2. Create LM scatterplots (per CellProfiler feature) comparing healthy versus failing cells without treatment and heart 7 cells treatment or no treatment.

I also include a notebook to generate plate map figures for all plates in the repository. I had to reprocess the normalization of plate 4 to normalize using the heart 7 DMSO-treated cells.

### CODE REVIEW

File: 3.process_cfret_features/scripts/1.single_cell_processing.py [hunk 1]
```diff
@@ -95,14 +95,23 @@
 
     # Save the modified DataFrame back to the same location
     annotated_df.to_parquet(output_annotated_file, index=False)
+    
+    # Only for Plate 4, we want to normalize to the DMSO treatments
+    if plate == "localhost231120090001":
+        samples = "Metadata_heart_number == 7 and Metadata_treatment == 'DMSO'"
+    # For all other plates, use the default
+    else:
+        samples = "all"
```
Comment (axiomcura) [inline 3.process_cfret_features/scripts/1.single_cell_processing.py:104]:
```suggestion
    # for all plates
     samples = "all"
     
    # Only for Plate 4, we want to normalize to the DMSO treatments
    if plate == "localhost231120090001":
        samples = "Metadata_heart_number == 7 and Metadata_treatment == 'DMSO'"
```
Some suggestion of making the code a lot cleaner and avoiding else clauses

File: 4.analyze_data/notebooks/linear_model/nbconverted/0.fit_linear_model_strat.py [hunk 2]
```diff
@@ -38,13 +44,13 @@
...
-specific_cell_types = ["failing", "healthy"]
+# Filter by cell type and only cells without DMSO treatment
+specific_type = [7]
+# specific_cell_types = ["Failing", "Healthy"]
 
 filtered_df = data_df[
-    (data_df['Metadata_treatment'].isin(specific_type)) &
-    (data_df['Metadata_cell_type'].isin(specific_cell_types))
+    (data_df['Metadata_heart_number'].isin(specific_type))
+    # (data_df['Metadata_cell_type'].isin(specific_cell_types))
```
Comment (axiomcura) [inline 4.analyze_data/notebooks/linear_model/nbconverted/0.fit_linear_model_strat.py:53]: Consider removing commented code if you plan on no longer using it.

**Review by axiomcura — APPROVED** [review]
LGTM, minor comments! Nice job! :+1:

File: 3.process_cfret_features/scripts/1.single_cell_processing.py (hunk 1 above)
Comment (jenna-tomkinson) [reply 3.process_cfret_features/scripts/1.single_cell_processing.py:104]: Love this! Just implemented!

File: 4.analyze_data/notebooks/linear_model/nbconverted/0.fit_linear_model_strat.py (hunk 2 above)
Comment (jenna-tomkinson) [reply 4.analyze_data/notebooks/linear_model/nbconverted/0.fit_linear_model_strat.py]: Just changed this when re-running the code, thanks!

**Comment by jenna-tomkinson** [comment]
@axiomcura Thank you for the review! I will be merging this in now!

**Pull request merged**